### DISCUSSION TITLE
Microsoft Edge keeps "hijacking" PDFs?

### DISCUSSION THREAD

**Discussion by Ander3232** [sumatrapdfreader/sumatrapdf#2707] [Q&A]
Hey guys,

Sumatra is great. However, since a recent Windows 10 update, Microsoft Edge has made itself my default PDF viewer, and I can't seem to set it back to Sumatra.

Windows' help says:

1. Open Settings (Win+G) and go to "Default Apps".
2. At the bottom of the list, click "Set default apps by file type".
3. In the list that appears, scroll down to the ".pdf" extension, click the default app, and select the app you want to use.

But when I try this, the window closes, and it has no effect. Any ideas? Thanks!

**Comment by GitHubRulesOK** [comment]
Varies from time to time, and MS use of Edge as PDF default, thus blocks SumatraPDF from setting itself as the default for PDF.

Resetting should work, until as you say some related MS update fettling.

Usually I don't seem to need to as I use both, but select any pdf and then replace handler with exe should work
 
instructions you quote and per https://www.sumatrapdfreader.org/docs/Set-as-default-PDF-viewer pick on an existing historic icon you need a fresh link

**Comment by Ander3232** [comment]
Thanks! I'd used the method you described, but apparently it was a registry change that didn't take effect till I'd restarted Windows. You know how it is—sometimes you have to restart, sometimes you don't... It's the Redmond Mystique (LOL). (I have friends running Linux who only have to restart when they replace hardware.) Cheers!

**Reply by GitHubRulesOK** [reply]
Tech Support Question 1 "Did you try switch off at the wall and count to 10 ?"

**Comment by Ander3232** [comment]
Or [as they often said in _The IT Crowd_](https://www.youtube.com/watch?v=t2F1rFmyQmY), "Have you tried turning it off and on again?"  😄

**Comment by Ander3232** [comment]
> Varies from time to time, and MS use of Edge as PDF default, thus blocks SumatraPDF from setting itself as the default for PDF.
> Resetting should work, until as you say some related MS update fettling.
> Usually I don't seem to need to as I use both, but select any pdf and then replace handler with exe should work
> instructions you quote and per https://www.sumatrapdfreader.org/docs/Set-as-default-PDF-viewer pick on an existing historic icon you need a fresh link

Months later—and none of this works now. Probably more Windows Updates changing things... That's usually what causes problems like these.

SumatraPDF doesn't even appear in the "Open with > Select another app" list now. (I can manually navigate to it and select it to open _that_ file in it, but the default isn't changed.) And if I go to Windows' own "Settings > Default Programs > by file type" and select Sumatra, the window crashes, and the setting is unchanged. 

I can only assume Microsoft wants Edge to keep popping up, as most of what people do on PCs now involves browsers. I wish Bill Gates had stuck with his CD-ROM vision of the future. 😛

**Comment by Ander3232** [comment]
Update—success!

Don't install the 64-bit version of SumatraPDF, even if you have a 64-bit system. Install the 32-bit version, and Windows lets you set it as the default PDF app. (Right-click a file > Open With > "Choose another app" > select SumatraPDF and select "Always use this app".) Yay!

I have no idea why this fails with the x64 Sumatra, but there's always seemed to be some lingering flakiness in the 64-bit-app world... Go figure.

**Reply by GitHubRulesOK** [reply]
@Ander3232 
 using that same recommended method with 64  bit I have no problems other than any "natural" crashes in the pre-release (currently this month a few due to annotation handling by MuPDF)

see the suggestion in https://github.com/sumatrapdfreader/sumatrapdf/discussions/2404?notification_referrer_id=NT_kwDOAMKf4bMzMjI1NjI1NzQwOjEyNzU0OTEz#discussioncomment-3667797

**Reply by Ander3232** [reply]
Okay, then—if I'm following you, you're saying my solution would be useful only to people for whom the 64-bit version _didn't_ work. Even I can comprehend that. It's good to be clear about these things, though. 😄

**Comment by TairikuOokami** [comment]
I install Sumatra via winget, so it registers properly and it is running as x64.
As for Edge I use policies to make sure, it does not hijack PDF anymore.

`reg add "HKLM\Software\Policies\Microsoft\Edge" /v "AlwaysOpenPdfExternally" /t REG_DWORD /d "1" /f`
`reg add "HKLM\Software\Policies\Microsoft\Edge" /v "ShowPDFDefaultRecommendationsEnabled" /t REG_DWORD /d "0" /f`

**Reply by Sieboldianus** [reply]
This doesn't seem to work anymore. I have these set and multiple times each day, microsoft-edge hijacks my default PDF link I set to sumatra-pdf. I don't even need to be working with PDF, edge appears to check every few minute and override my default. Crazy MS.

**Reply by TairikuOokami** [reply]
Maybe it depends on Windows version or installed apps, MS365 loves to grab PDF files.

<img width="1920" height="1080" alt="capture_06032026_153054" src="https://github.com/user-attachments/assets/9cc5a955-86f0-4f1f-9d88-597958035000" />

**Reply by Sieboldianus** [reply]
I checked and this seems not the case. Today is the fourth day that edge has completely hijacked my PDF associations, with apparently no way back 
<img width="1484" height="820" alt="grafik" src="https://github.com/user-attachments/assets/c31fc6bd-faba-4587-b3d0-7e484e4c85ec" />

**Reply by Sieboldianus** [reply]
I had to do the following:
Elevated CMD:
```
schtasks /change /disable /tn "\Microsoft\Windows\AppxDeploymentClient\UCPD velocity"
sc config UCPD start=disabled
```
- this disables a protection driver from microsoft that prevents deleting UserChoice folder in regedit, where microsoft-edge was written
- restart

Elevated regedit:
```
Computer\HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Explorer\FileExts\.pdf\
```
-> Delete the folder `UserChoice`

Afterwards, my SumatraPDF association stays. However, the UserChoice folder hasn't popped up yet again, so I will wait before activtaint again the velocity crap and UCPD that microsoft invented to protect their edge-pdf-hijack.

[Edit] Negative.. after a few minutes, UserChoice folder is back with `ProgID` MSEdgePDF...

**Reply by Sieboldianus** [reply]
I manually created the `UserChoice` folder now and set a Key `ProgId` with `SumatraPDF` as value manually. I then blocked the system from changing/modifying this key via custom access rules on the registry folder. So far it seems to work.

[edit] 3 days later: this did also not work. 

I now tried this:

```
Computer\HKEY_LOCAL_MACHINE\SOFTWARE\Classes\MSEdgePDF\shell\open\command
```
-> Override the path to `msedge.exe` to the path to `SumatraPDF.exe` (`"C:\Program Files\SumatraPDF\SumatraPDF.exe" "%1"`)

...and then edit this Registry Key, to update the icon:
```
MSEdgePDF\DefaultIcon
```

The "Standard" key needs to have the value: `"C:\Program Files\SumatraPDF\SumatraPDF.exe",1` (e.g.)
